.github/workflows/rust.yml

```diff
@@ -3,28 +3,58 @@ name: Rust
 on: [push]
```

Comment:
It's possible to add the `pull_request` event, however I generally don't see this. I think that's because [`push` should get triggered](https://help.github.com/en/actions/reference/events-that-trigger-workflows#push-event-push) on the fork's branch unless this is explicitly disabled in the repo's settings.

So I'm happy to add it but it may mean the workflow would be run twice.

Reply:
Not sure I understand.. Are you supposed to go to the fork from the PR in order to see the status of the CI run? That's inconvenient for sure, and easy to forget.

Keeping the current behavior is fine by me, as we're not turning Travis off yet.

Reply:
Basically the `push` action is more general than the `pull_request` action. So when I created this PR for example, that there is a workflow run for that. Likewise, each time I push this branch, there's a workflow run. I think it's probably redundant to subscribe to both, but I'm not entirely sure. I haven't really seen many examples that seem to use both, but it's trivial to add.

Reply:
The run on this PR is from me pushing it to a branch on this repo: https://github.com/andygrove/sqlparser-rs/tree/fixup/code-coverage

If you look on other PRs like [#187](https://github.com/andygrove/sqlparser-rs/pull/187), only Travis runs on them, though we have had a GitHub action with a `push` handler for a while.

Do you know of another project, which runs CI on pull requests via Actions?

Reply:
When you create a PR there should be a new PR ref created by GH. I guess what I don't know is if this branch creation triggers the `push` action?

That said and to answer your question, yes we use Actions throughout a number of projects at work for both CI and CD (and really like it, even over CircleCI, for some projects. [This is one example](https://github.com/curology/vicuna/blob/master/.github/workflows/rust.yml) of how we have things set up. Note that we only subscribe to `push`. That said, we do tend to create feature branches instead of forks, so maybe that's why that hasn't been an issue?

I'm more than happy to add it. I did read a few random comments that seemed to suggest having both could lead to duplicate workflow runs but I'm not sure if that's still the case or even entirely accurate.

Reply:
Yep, if you create the feature branches on the main repo, that explains why the `push` event is enough. In sqlparser-rs 99% of the PRs come from an outside fork, so I think a pull-request event makes sense. I'm still not sure what is the problem with duplicate runs, but I guess we can try it and see.

Reply:
Agree--it's extremely easy to add or remove. I'll add it here.